src/Tasks/CurlLinkChecker.php

```diff
@@ -66,11 +84,28 @@ public function checkLink($href)
         // No cached result so just request
         $handle = curl_init($href);
         curl_setopt($handle, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($handle, CURLOPT_CONNECTTIMEOUT, 5);
+        curl_setopt($handle, CURLOPT_TIMEOUT, 10);
         if ($this->config()->get('follow_location')) {
             curl_setopt($handle, CURLOPT_FOLLOWLOCATION, true);
         }
-        curl_setopt($handle, CURLOPT_CONNECTTIMEOUT, 5);
-        curl_setopt($handle, CURLOPT_TIMEOUT, 10);
+
+        // Add user agent
+        $userAgent = trim($this->config()->get('user_agent'));
+        if ($userAgent) {
+            curl_setopt($handle, CURLOPT_USERAGENT , $userAgent);
+        }
+
+        // Other headers
+        if ($headers = $this->config()->get('headers')) {
+            if (is_array($headers)) {
+                 curl_setopt($handle, CURLOPT_HTTPHEADER , $headers);
+            } else {
+                 curl_setopt($handle, CURLOPT_HTTPHEADER , array($headers));
+            }
```

Comment:
You could do `$headers = (array) $this->config()->get('headers')` and remove the condition here.